py_zipkin/logging_helper.py

```diff
@@ -359,14 +338,50 @@ def add_span(
         binary_annotations,
         timestamp_s,
         duration_s,
+        endpoint,
+        sa_endpoint,
     ):
+        thrift_endpoint = thrift.create_endpoint(
+            endpoint.port,
+            endpoint.service_name,
+            endpoint.ipv4,
+            endpoint.ipv6,
+        )
+
+        thrift_annotations = thrift.annotation_list_builder(
+            annotations,
+            thrift_endpoint,
+        )
+
+        # Binary annotations can be set through debug messages or the
+        # set_extra_binary_annotations registry setting.
+        thrift_binary_annotations = thrift.binary_annotation_list_builder(
+            binary_annotations,
+            thrift_endpoint,
+        )
+
+        # Add sa binary annotation
+        if sa_endpoint is not None:
```

Comment:
maybe pull this out into its own function?

Reply:
I don't see much value in it being a separate functions since it's not something we can reuse. Also it'd be a function that changes the content of one of its arguments rather than return the result which I don't like too much.